### ISSUE TITLE
Automatic support or clearer docs for tcl-tk on MacOS

### ISSUE THREAD

**Issue by asfaltboy** [pyenv/pyenv#1545]
### Description
I am having issues on MacOS Catalina with homebrew tcl-tk lib not being picked up. While the default 8.5 shipped with MacOS is unsupported, as per [python's docs][1]. The error I see is:

```
> python3.8 -m tkinter -c 'tkinter._test()'
DEPRECATION WARNING: The system version of Tk is deprecated and may be removed in a future release. Please don't rely on it. Set TK_SILENCE_DEPRECATION=1 to suppress this warning.
Traceback (most recent call last):
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/tkinter/__main__.py", line 7, in <module>
    main()
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/tkinter/__init__.py", line 4552, in _test
    root = Tk()
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/tkinter/__init__.py", line 2263, in __init__
    self._loadtk()
  File "/Users/pavel.savchenko/.pyenv/versions/3.8.0/lib/python3.8/tkinter/__init__.py", line 2279, in _loadtk
    raise RuntimeError("tk.h version (%s) doesn't match libtk.a version (%s)"
RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
```

Should pyenv attempt to detect it similarly to how we do with [`use_homebrew_readline`][2]?

Alternatively, we could also add the example from #1397 to the common build problems wiki page. I.e this works for me: 
```
PYTHON_CONFIGURE_OPTS="--with-tcltk-includes='-I/usr/local/opt/tcl-tk/include' \
  --with-tcltk-libs='-L/usr/local/opt/tcl-tk/lib -ltcl8.6 -ltk8.6'"\
   pyenv install 3.8.0
```

Please let me know which is preferable, and I can submit the appropriate PR

### Environment
- Platform information: MacOS Catalina 10.15.3
- OS architecture: amd64
- pyenv version: 1.2.16
- Python version: 3.8.0
-  C Compiler information: Apple clang version 11.0.0 (clang-1100.0.33.17)

[1]: https://www.python.org/download/mac/tcltk/
[2]: https://github.com/grahamannett/pyenv/blob/master/plugins/python-build/bin/python-build#L1671-L1681

**Comment by dggsax** [comment]
Curious about this as well, the solution offered by Carl on this issue (#94 ) worked for me

**Comment by weaming** [comment]
> Curious about this as well, the solution offered by Carl on this issue (#94 ) worked for me

Works also https://github.com/pyenv/pyenv/issues/1375#issuecomment-524280004

**Comment by emeti** [comment]
I hope I understand @asfaltboy , here is my view of this 

Contex
-------
- From https://www.python.org/download/mac/tcltk/#built-in-8-6-8  "All current python.org installers for Python 3.7.x, 3.6.x, and 2.7.x link to their own built-in Tcl/Tk 8.6 frameworks and do not use external Tcl/Tk frameworks". 

- Building CPython from sources on MacOS with tcl-tk installed via brew fails if the default configure options are used. It would not detect and pick up homebrew installed tcl/tk 8.6 and requires using --with-tcltk-includes , --with-tcl-tk-libs in the build.

Choice
-------
To handle tcl-tk problem by pyenv there are two options:
1. use hombrew's tcl-tk and build with it by default
2. allow pyenv install to fail and let the user figure out the build options or environment variable setting, pointing to documentation plus an error message with clear instructions

The question is - are pyenv users knowledgable enough to handle option 2 or should it "just work"? 

I would vote for the latter for my own use of pyenv but I think "general public" would probably prefer smooth and problem free installs (i.e. option 1).

tkinter existence in the python standard library is the crux of the problem (python rant)
-----
It is unfortunate that tkinter is part of python standard library and not an external module. Forcing external gui framework dependency into the low layer of the stack (and shipping by python as built-in a gui framework that I do not need or want, and obviously Apple also does not care about, why would they?) is the reason of all those problems. This smells of monolithic non-portable entanglement. If tkinter was an external module all those issues would disappear, or be in the proper place - left to handle to those who really caused them.

The deferred PEP 534 https://www.python.org/dev/peps/pep-0534/ is interesting. I would prefer to have python builds without tcltk and idle. Cpython currently does not offer a configure option --without-tcltk, unfortunately

**Comment by asfaltboy** [comment]
Thanks, @emeti, that summary is on point 👍 . It seems that shipping Tcl/Tk with a pre-build OS-specific installer seems like a compromise the core team had to make. It seems the main goal of support Tkinter is to allow IDLE the beginner-friendly IDE for python, to run out of the box.

The suggestion to remove it isn't very new (see this post from [Petr Viktorin][1]). And [PEP 594][2] is a step in the right direction, due to how Tkinter is used today, I believe it's not considered a "Dead Battery".

If you'd like to share your views on this cleanup, check out [the discussion thread for the PEP][3].

[1]: https://mail.python.org/archives/list/distutils-sig@python.org/message/GYB4NRWDU5SEWI2X47LSNDER7IGAWMMB/
[2]: https://www.python.org/dev/peps/pep-0594/
[3]: https://discuss.python.org/t/pep-594-removing-dead-batteries-from-the-standard-library/1704/64

**Comment by emeti** [comment]
@asfaltboy , thanks for the PEP 594 reference, some good points there, and I agree that the discussion of python stdlib scope belongs in the python territory. 

IMHO tkinter is not a "dead battery" (very much alive for IDLE and tcl/tk lovers) but it is rather a "welded" one (included and preinstalled) and not necessarily good or needed for many python use cases, where gui or tk are not required. The choice is between "one size fits all" and "select what you need". Linux distributions have resolved it in favour of the latter by providing python and python-tk or python-tkinter as separate packages.

+1 for following in pyenv in the main definition similarly with what upstream does on MacOS, i.e. smoothly detect and use tcl-tk 8.6 from homebrew, that way pyenv python version is close to python.org release and trouble free for novice and non-expert users (is that the main audience of pyenv though?).

I am unsure of one thing ... if the suggested change would enforce using the homebrew's tcl-tk with every build (I hope not) or only configure the build to link with it, if it is already present, i.e. there would still be a way to build python --without-tk on MacOS ? I understand that is what currently happens, i.e. pyenv python builds of python interpreter work fine on "tk-less" systems (I have just tested that is the case on a fedora 32 without tk, the tk-devel is not the absolutely required dependency although it is [a pyenv suggested requirement](https://github.com/pyenv/pyenv/wiki#suggested-build-environment) ),  an only users may find out later that it is a missing dependency when they try to use tkinter and get "Traceback ... runpy.py ... import _tkinter # If this fails your Python may not be configured for Tk". Sorry if that is obvious, I am a pyenv user (and really like it) and an observer and do not fully understand what is going on in the python-build script and the definitions.

**Comment by Diegus83** [comment]
Is there an accepted solution or workaround for this?

I reinstalled pyenv today using pyenv installer, running macOS 10.4.6.

I tried the steps outlined in this post https://stackoverflow.com/questions/60469202/unable-to-install-tkinter-with-pyenv-pythons-on-macos/60469203#60469203

Then I reinstalled python 3.9.1 with pyenv and set it as global.

When I check I still get version 8.5 for tkinter, and the text is missing from the buttons in the code I'm trying to run.
